### PR TITLE
Backend paddle: modify fnn

### PR DESCRIPTION
**Pull request by lijialin03**
1. add regularizers
2. add dropout

### CODE REVIEW

File: deepxde/nn/paddle/fnn.py [hunk 1]
```diff
@@ -37,6 +48,8 @@ def forward(self, inputs):
                 if isinstance(self.activation, list)
                 else self.activation(linear(x))
             )
+        if self.dropout is not None:
```
Comment (lululxvi) [inline deepxde/nn/paddle/fnn.py:51]: Is this correct? Dropout should be inside the for loop.

Comment (lijialin03) [reply deepxde/nn/paddle/fnn.py]: No it is not, I have updated code.
What is more, I noticed that PFNN in fnn.py has `regularization`. So I added it in this PR but not open a new one now that code should be changed only one line.

File: deepxde/nn/paddle/fnn.py [hunk 2]
```diff
@@ -20,6 +28,13 @@ def __init__(self, layer_sizes, activation, kernel_initializer):
             self.activation = activations.get(activation)
         initializer = initializers.get(kernel_initializer)
         initializer_zero = initializers.get("zeros")
+        self.regularizer = regularizers.get(regularization)
+        self.dropout_rate = dropout_rate
+        self.dropouts = [
+            paddle.nn.Dropout(p=dropout_rate)
+            for _ in range(1, len(layer_sizes) - 1)
+            if dropout_rate > 0.0
+        ]
```
Comment (lululxvi) [inline deepxde/nn/paddle/fnn.py:37]:
```
if dropout_rate > 0:
    self.dropouts = [paddle.nn.Dropout(p=dropout_rate) for _ in range(1, len(layer_sizes) - 1)]
```

File: deepxde/nn/paddle/fnn.py [hunk 3]
```diff
@@ -37,6 +52,8 @@ def forward(self, inputs):
                 if isinstance(self.activation, list)
                 else self.activation(linear(x))
             )
+            if self.dropout_rate > 0.0:
```
Comment (lululxvi) [inline deepxde/nn/paddle/fnn.py:55]: `if self.dropout_rate > 0:`

File: deepxde/nn/paddle/fnn.py [hunk 4]
```diff
@@ -3,12 +3,20 @@
...
     """Fully-connected neural network."""
 
-    def __init__(self, layer_sizes, activation, kernel_initializer):
+    def __init__(
+        self,
+        layer_sizes,
+        activation,
+        kernel_initializer,
+        regularization=None,
+        dropout_rate=0.0,
```
Comment (lululxvi) [inline deepxde/nn/paddle/fnn.py:18]: `dropout_rate=0,`
